packages/@aws-cdk/integ-tests-alpha/lib/assertions/providers/lambda-handler/utils.ts

```diff
@@ -1,11 +1,12 @@
-export function parseJsonPayload(payload: any): any {
-  // sdk v3 returns payloads in Uint8Array
-  // If the payload is not convertible to a buffer or parsable as JSON, we don't touch it
+export function parseJsonPayload(payload: string | Buffer | Uint8Array | undefined | null): any {
+  // sdk v3 returns payloads in Uint8Array, either it or a string or Buffer
+  // can be cast into a buffer and then decoded.
+  const text = new TextDecoder().decode(Buffer.from(payload ?? ''));
+  if (!text) { return { }; }
   try {
-    const buffer = Buffer.from(payload);
-    return JSON.parse(new TextDecoder().decode(buffer));
+    return JSON.parse(text);
   } catch {
-    return payload;
+    throw new Error(`return values from user-handlers must be JSON objects. got: "${text}"`);
```

Comment:
Can a Lambda Handler not return text? Or is it always a JSON parsable string, i.e. `"foobar"`?

Reply:
Looks to be always a JSON representation of a JavaScript object.

Reply:
I.e. a plain text is returned as `"Hi I'm a plain string"`.

Reply:
I'm still concerned we are going to fail on valid values or return unexpected values.
In the assertion handler `parseJsonPayload` is naively called on a potentially existing `payload` property of the response object. That's different than the provider framework where we know we are dealing with a lambda response.

In hindsight, maybe we should do the assertions handler on a different PR.

Reply:
I agree that the code is making too many assumptions, and might be subtly broken.

I would posit that it is subtly broken already, and that nothing we're doing is making it worse than what it currently is.

But sure, I'll revert!